flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPool.java

```diff
@@ -470,7 +467,7 @@ private MemorySegment requestOverdraftMemorySegmentFromGlobal() {
 
         MemorySegment segment = networkBufferPool.requestPooledMemorySegment();
         if (segment != null) {
-            numberOfRequestedOverdraftMemorySegments++;
+            numberOfRequestedMemorySegments++;
```

Comment:
I think It is better if we increase/decrease numberOfRequestedMemorySegments only in one place to avoid future mistakes. Right now we have `requestMemorySegmentFromGlobal` and `requestOverdraftMemorySegmentFromGlobal`. Perhaps, it makes sense to extract common code from them, etc.